hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/ObjectEndpoint.java

```diff
@@ -196,6 +200,17 @@ public Response put(
 
       // Normal put object
       OzoneBucket bucket = volume.getBucket(bucketName);
+      if (length == 0 &&
+          ozoneConfiguration
+              .getBoolean(OZONE_S3G_FSO_DIRECTORY_CREATION_ENABLED,
+                  OZONE_S3G_FSO_DIRECTORY_CREATION_ENABLED_DEFAULT) &&
+          bucket.getBucketLayout() == BucketLayout.FILE_SYSTEM_OPTIMIZED) {
+        s3GAction = S3GAction.CREATE_DIRECTORY;
+        // create directory
+        getClientProtocol()
+            .createDirectory(volume.getName(), bucketName, keyPath);
+        return Response.ok().status(HttpStatus.SC_OK).build();
+      }
```

Comment:
Thank @mladjan-gadzic .  Minor change.  This added block above checking the length, FSO layout and handling the fso directory create (L203-213), should be placed after the block below handling the copy header and CopyObject L217-226.  After re-run the CI workflow.